components/vision.py

```diff
@@ -12,4 +14,13 @@ def get_vision_data(self) -> Tuple[float, float, float]:
         """Returns a tuple containing the distance (metres), angle (radians), and timestamp (time.monotonic)
         If it can't get info, it returns [None, None, None]
         """
-        return self.entry.getDoubleArray([None, None, None])
+        self.data = self.entry.getDoubleArray([None, None, None])
+        return self.data
+
+    def is_ready(self) -> int:
+        if abs(self.data[2] - time.monotonic()) > 0.5:
```

Comment:
This... doesn't seem correct. How are you guaranteeing the clock on the RPi is the same as the clock on the RIO?

Reply:
i have now changed it